### PR TITLE
🧼 remove v4.44 deprecations

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

See title :)

@zucchini-nlp -> video llava changes
@SunMarc -> all other changes

### CODE REVIEW

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -370,92 +370,6 @@ def check_support_param_buffer_assignment(model_to_load, state_dict, start_prefi
...
-        state_dict (`Dict[str, torch.Tensor]`): The state dictionary of a model to save.
-        max_shard_size (`int` or `str`, *optional*, defaults to `"10GB"`):
-            The maximum size of each sub-checkpoint. If expressed as a string, needs to be digits followed by a unit
-            (like `"5MB"`).
-        weights_name (`str`, *optional*, defaults to `"pytorch_model.bin"`):
-            The name of the model save file.
-    """
-    logger.warning(
-        "Note that `shard_checkpoint` is deprecated and will be removed in v4.44. We recommend you using "
-        "split_torch_state_dict_into_shards from huggingface_hub library"
```
Comment (gante) [inline src/transformers/modeling_utils.py:402]: (followed this instructions whenever `shard_checkpoint` was used)

**Review by zucchini-nlp — CHANGES_REQUESTED** [review]
Thanks for fixing but the processing logic refactor wen longer than I expected. We are still doing the last bits in https://github.com/huggingface/transformers/pull/33424/ and the checkpoints on the hub are not updated yet (tracker https://github.com/huggingface/transformers/issues/33374). So we can't be raising ValueError yet

Maybe we can change the deprecation version to be after a few more minor releases. We have also same warnings in all llava and blip models afair. Also now I think we can even do more than just a few releases, since the changes will invalidate many models on the hub that are not owned by us. WDYT, which version we can put in the message?

**Comment by gante** [comment]
> since the changes will invalidate many models on the hub that are not owned by us.
We should then support it for as long as possible :) If it is critical to any popular model, I would not even set a deprecation version target at all. Leave the exception without target, and set a target when it blocks you from doing something OR if it is causing many issues. An example of this is supporting `model.config` modifications to parameterize `generate` -- it is super inconvenient for maintenance, but many examples/models rely on it. 

If it is only used in some niche models, and applying the deprecation (raising exception) is a significant win for transformers, then I would set at least 6 minor versions.

LMK which one is it, and I will update the warning.

**Comment by zucchini-nlp** [comment]
I hope you meant `warning`, not `exception`. I don't think we should be raising exception unless at least the official checkpoint is updated accordingly. 

I would say this particular one is a niche model, given download stats from the hub. And most older llava and BLIP models may be outdated fast as we get newer and better releases each month. But I think we still can't enforce users to do something by elevating this to an Exception. My preference is to keep it for 6 minor versions as a warning, and try to update official models on the hub until that. We have write access to all of them afaik except for Video-LLaVA so updating will be straightforward

File: src/transformers/modeling_utils.py (hunk 1 above)
Comment (SunMarc) [reply src/transformers/modeling_utils.py:402]: Unfortunately, this is not the exact function. We can just replace it directly. You need to build the shard and the index again using this function. Check the example here: https://huggingface.co/docs/huggingface_hub/main/en/package_reference/serialization#huggingface_hub.split_torch_state_dict_into_shards.example

**Review by SunMarc — COMMENTED** [review]
Thanks for the update ! I left a comment. Also, I check that we need to enforce huggingface-hub to 0.24.0 because of this [commit](https://github.com/huggingface/huggingface_hub/releases?page=2#:~:text=Support%20max_shard_size%20as%20string%20in%20split_state_dict_into_shards_factory%20by%20%40SunMarc%20in%20%232286).  we accept `max_shard_size` with string format only from this version.

**Comment by gante** [comment]
> I hope you meant `warning`, not `exception`

@zucchini-nlp haha yes, `warning`! In essence, allowing the behavior.

**Comment by gante** [comment]
@zucchini-nlp @SunMarc PR comments addressed, LMK if they look good to you :)

**Review by zucchini-nlp — APPROVED** [review]
Thanks, looks good to me! Can we propagate the same thing to all files with this deprecation warning? I guess it is all LLaVA and some of BLIP models (processing and modeling files)

**Review by SunMarc — APPROVED** [review]
Nice ! Thanks for iterating !

**Review by LysandreJik — APPROVED** [review]
Ok, sounds good! Thanks for the PR and cleanup @gante

**Review by ArthurZucker — APPROVED** [review]
🧼 thanks

**Comment by SunMarc** [comment]
To pass the CI, you need to rebase the PR @gante  ;)

**Pull request merged**